### ISSUE TITLE
Open MPI 4.1.2 static-only build reachable_netlink linking issue on Ubuntu 20.04

### ISSUE THREAD

**Issue by dqwu** [open-mpi/ompi#10056]
This issue can be reproduced with Open MPI 4.1.2 release on Ubuntu 20.04.
Not reproducible with a shared build. Not reproducible on Ubuntu 18.04.
Looks like a linking issue related to reachable_netlink lib on Ubuntu 20.04.

Steps to reproduce
```
wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.2.tar.gz
tar zxf openmpi-4.1.2.tar.gz
cd openmpi-4.1.2

./configure --prefix=/path/to/openmpi/installation --enable-static --disable-shared
make -j8
make install

export PATH=/path/to/openmpi/installation/bin:$PATH

cat <<EOF >> test_mpi.c
#include <mpi.h>
int main(int argc, char *argv[])
{
  MPI_Init(&argc, &argv);
  MPI_Finalize();

  return 0;
}
EOF

mpicc test_mpi.c
```
Output:
```
/usr/bin/ld: /path/to/openmpi/installation/lib/libopen-pal.a(reachable_netlink_utils_common.o): undefined reference to symbol 'nlmsg_set_proto'
/usr/bin/ld: /lib/x86_64-linux-gnu/libnl-3.so.200: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

One possible workaround:
Configure Open MPI without reachable-netlink support (--enable-mca-no-build=reachable-netlink)

**Comment by bwbarrett** [comment]
Thank you for bringing this to our attention.  I can confirm that there is a bug in the configure logic for the netlink reachable component such that (for static library builds) we are not adding the netlink libraries to the wrapper compilers.  I am working on a patch to fix this issue and it will be fixed in the upcoming 4.1.3 release.

As you note, `--enable-mca-no-build=reachable-netlink` will work around this issue by not creating a dependency on the netlink librarys from libmpi.  Another option is to add `--with-wrapper-libs="-lnl-3 -lnl-route-3"` to the configure command.  Unless the machines used for your application have a particularly complicated network setup with multiple interfaces per machine, the reachable component is likely overkill and either option is equally valid.

**Comment by bwbarrett** [comment]
#10107 fixes the issue on master; more backports to fix some bugs in opal_mca.m4 will be required on v4.1.x

**Comment by bwbarrett** [comment]
I was wrong; https://github.com/open-mpi/ompi/pull/10130 was all that was needed to fix this on 4.1.x.
